### ISSUE TITLE
Is diff pager supported?

### ISSUE THREAD

**Issue by WindSoilder** [speedyleion/gh-difftool#220]
Hi, thank you for the extension, I found it very interesting.
I'm trying to using this with [difftastic](https://github.com/Wilfred/difftastic), here is my `.gitconfig` file(I just apply these config from `delta` and `difftastic` instructions):
```ini
[pager]
    difftool = true

[diff]
    tool = difftastic

[difftool]
    prompt = false

[difftool "difftastic"]
    cmd = difft "$LOCAL" "$REMOTE"

[difftool.difftastic]
    path = /Users/WindSoilder/.cargo/bin/difft
```
Then I run `PAGER=less gh difftool <pr_num>`, it uses `difftastic` properly, but without paging.  Is there any way to support paging?

**Comment by speedyleion** [comment]
Finally got a chance to look at this a bit.
I'll have to chew on this a bit to see if paging is something that `gh-difftool` should be directly supported or not.
The initial drive was GUI tools, as `gh pr diff <PR>` was available. However difftastic has a much nicer output than that.
Also I have no idea how to handle paging so that will be something to look into as well. Currently `gh-difftool` launches the difftool for each individual file and I'm not sure how/where to hook up paging there or if it would need to be reworked in general.

A work around, using a wrapper script, https://github.com/speedyleion/gh-difftool?tab=readme-ov-file#unsupported-difftools

The wrapper script `foo.sh`
> Note the `--color=always` being used with difft
```shell
#!/bin/sh
/path/to/difft --color=always $1 $2
```
Then in one's git config
```ini
[difftool.difftastic]
    path = /path/to/the/wrapper/foo.sh
```

The command
```shell
gh difftool --tool difftastic <PR> | less -R
```

**Comment by WindSoilder** [comment]
Thanks!  I will take the workaround, it works well for me :-D

**Comment by speedyleion** [comment]
TL;DR I think `gh-difftool` shouldn't directly support paging, because it appears that `git-difftool` doesn't support it.
Adding support for `config.difftool.<tool>.cmd` might be an improvement here.
If I add the following to my gitconfig, then `git-difftool` supports paging with difftastic (but only on a file per file basis)
```
[difftool.difftastic]
    cmd = /path/to/difft $LOCAL $REMOTE --color=always | less -RFX
```

Adding some discovery notes.
I found, https://docs.rs/pager/latest/pager/, which provides a fairly simple solution to enable paging.
It does require a slight modification to the async behavior in `main()`, see diff below.
This will set the pager, which forks the process, and then creates the tokio runtime.
In this example i used `less -FRX`, but an ideal solution would use the git logic for pager handling, https://git-scm.com/docs/git-config#Documentation/git-config.txt-corepager
```diff
-#[tokio::main]
-async fn main() -> Result<()> {
+fn main() -> Result<()> {
     let cli = Cli::parse();
+    Pager::with_pager("less -FRX").setup();
+
+    let runtime = tokio::runtime::Runtime::new()?;

     let mut gh = gh_interface::GhCli::new(Command::new("gh"));
     let mut pr = match cli.pr {
@@ -106,7 +109,7 @@ async fn main() -> Result<()> {

     // Important, do this after the name only check as name only doesn't need a difftool
     let difftool = git_config::Difftool::new(std::env::current_dir()?, cli.tool.as_deref())?;
-    diff(difftool, change_set).await?;
+    runtime.block_on(diff(difftool, change_set))?;
     Ok(())
 }
 ```
 
 This unfortunately is still lacking:
 - it breaks vimdiff with `Vim: Warning: Output is not to a terminal`
 - difftastic still doesn't provide colors unless a wrapper script is used with `--color=always`
 
 I'm guessing it's for things like this vimdiff error why `git-difftool` doesn't do paging
 Interestingly there is now also a [`guitool`](https://git-scm.com/docs/git-difftool#Documentation/git-difftool.txt-diffguitool) option (or maybe I glossed over it before). I haven't quite figured out the difference other than it's looked at first if `--gui` is used or is set in the config

**Comment by WindSoilder** [comment]
So it seems that a wrapper script is a good solution for now.
Maybe we can add some information like this in Readme.

**Comment by speedyleion** [comment]
Added support for `cmd` in #221 

```ini
[difftool.difftastic]
    cmd = /path/to/difft --color=always $LOCAL $REMOTE
```

This is released in https://github.com/speedyleion/gh-difftool/releases/tag/v1.2.0
